src/main/java/hudson/remoting/SynchronousCommandTransport.java

```diff
@@ -17,6 +18,15 @@
 public abstract class SynchronousCommandTransport extends CommandTransport {
     protected Channel channel;
 
+    private static final String RDR_SOCKET_TIMEOUT_PROPERTY_NAME=SynchronousCommandTransport.class.getName() + ".failOnSocketTimeoutInReader";
+    
+    /**
+     * Enables the original aggressive behavior, when the channel reader gets 
+     * interrupted on any {@link SocketTimeoutException}.
+     * @since TODO
+     */
+    private static boolean RDR_FAIL_ON_SOCKET_TIMEOUT = Boolean.getBoolean(RDR_SOCKET_TIMEOUT_PROPERTY_NAME);
```

Comment:
It's remoting, nevermind... So library also contributing more and more system variables :)